src/App/Router/index.tsx

```diff
@@ -0,0 +1,84 @@
+import React, { useEffect } from 'react';
+import { Route, Routes, useLocation, useNavigate } from 'react-router-dom';
+import { isFeatureEnabled } from 'utilities';
+
+import 'assets/styles/App.scss';
+import { routes } from 'constants/routing';
+import { useAuth } from 'context/AuthContext';
+import Account from 'pages/Account';
+import ConvertVrt from 'pages/ConvertVrt';
+import Dashboard from 'pages/Dashboard';
+import Vote from 'pages/Governance';
+import History from 'pages/History';
+import IsolatedPools from 'pages/IsolatedPools';
+import { CorePoolMarket, IsolatedPoolMarket } from 'pages/Market';
+import { CorePool, IsolatedPool } from 'pages/Pool';
+import Proposal from 'pages/Proposal';
+import Swap from 'pages/Swap';
+import Vai from 'pages/Vai';
+import Vaults from 'pages/Vault';
+import Voter from 'pages/Voter';
+import VoterLeaderboard from 'pages/VoterLeaderboard';
+import Xvs from 'pages/Xvs';
+
+const Router = () => {
+  const { accountAddress } = useAuth();
+  const location = useLocation();
+  const navigate = useNavigate();
+
+  // Redirect to account page if user has already connected their wallet and is
+  // visiting the dashboard. If they refresh the page while being on the
+  // dashboard, the redirection will not happen
+  useEffect(() => {
+    if (!!accountAddress && location.pathname === routes.dashboard.path) {
+      navigate(routes.account.path);
+    }
+  }, [location, accountAddress]);
```

Comment:
This is causing an issue where if you connect your wallet to the frontend you get redirected to the Account page, then if you try to go to the Dashboard page you always get redirected to the Account page.

The original behavior was that if you start a new session by visiting the Dashboard while you already have your wallet connected to the UI, you get redirected to the Account page. After that you can navigate to any other page without getting redirected to the Account page. You also should only get redirected to the Account page at the start of the session if you were going to the Dashboard page: if you were going to any other page you don't get redirected.